src/components/container/MostRecentNotification/MostRecentNotification.tsx

```diff
@@ -38,31 +62,63 @@ export default function (props: MostRecentNotificationProps) {
 			parameters.type = props.notificationType;
 		}
 
-		MyDataHelps.queryNotifications(parameters).then(function (result) {
-			if (result.notifications.length) {
-				setNotification(result.notifications[0]);
+		if (!props.notificationIdentifierRegex) {
+			const notificationPage = await MyDataHelps.queryNotifications(parameters)
+			setLoading(false);
+			if (notificationPage.notifications.length > 0) {
+				setNotification(notificationPage.notifications[0]);
+			}
+		} else {
+			const notification = await findMatchingNotification(parameters, props.notificationIdentifierRegex.source);
+			setLoading(false);
+			if (notification) {
+				setNotification(notification);
 			}
-		});
+		}
 	}
 
-	useEffect(() => {
-		initialize();
-		MyDataHelps.on("applicationDidBecomeVisible", initialize);
-		return () => {
-			MyDataHelps.off("applicationDidBecomeVisible", initialize);
-		}
-	}, []);
+	async function findMatchingNotification(queryParams: NotificationQueryParameters, identifierRegex: string): Promise<Notification | null> {
```

Comment:
This line is returning a VSCode error for me:
![image](https://github.com/user-attachments/assets/3bf97aab-1bfb-456a-b7aa-e5f7b473738b)